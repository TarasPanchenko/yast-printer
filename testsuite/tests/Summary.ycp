/**
 * File:
 *  Summary.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Miscelaneous tests.
 *
 * Authors:
 *  Petr Blahos <pblahos@suse.cz>
 *
 * $Id$
 *
 */

{
    import "Pkg";

    include "testsuite.ycp";
    import "Printer";

    map READ = $["printerdb" : $[ "printername" : "- fake name -"],
		 "ppd" : $["file" : $["isppd" : false ]]];
    map WRITE = $[];
    map EXEC = $[];

    Printer::printcap = [
	$[ "name" : "lp", "unique_key" : "1", "uri" : "parallel:/dev/lp0", "vendor_ieee": "Canon", "device_ieee":"BJC-6100", "database" : "foomatic", ],
	$[ "name" : "color", "unique_key" : "1", "uri" : "parallel:/dev/lp0", "vendor_ieee": "Canon", "device_ieee":"BJC-6100", "database" : "foomatic",],
	$[ "name" : "mono", "unique_key" : "2", "uri" : "usb:/dev/usb/lp0", "vendor_ieee": "hewlett-packard", "device_ieee":"hp-laserjet-4", "database" : "foomatic",],
	$[ "name" : "hand", "uri" : "usb:/dev/usb/lp1", "vendor_ieee":"epson", "device_ieee": "generic-gdi", "database" : "foomatic",],
	];
    Printer::autodetected = [
	$["vendor":"Canon","device":"BJC-6100","unique_key":"1","uri":"parallel:/dev/lp0"],
	$["vendor":"Epson","device":"Stylus","unique_key":"3","uri":"usb:/dev/usb/lp1"],
	];

    TEST (``(Printer::Summary ([`short])), [READ, WRITE, EXEC], 0);
}
