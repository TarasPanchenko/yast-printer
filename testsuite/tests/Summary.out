Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Execute	.target.bash_output "locale -k charmap" $[]
Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Read	.target.size "" 0
Read	.ppd.file.isppd "" false
Read	.target.size "" 0
Read	.ppd.file.isppd "" false
Return	<ul><li><p>Canon BJC-6100: Parallel printer on /dev/lp0<br>Configured as lp, color.</p></li><li><p>Epson Stylus: USB printer on /dev/usb/lp1<br>Not configured yet.</p></li><li><p>epson generic-gdi: USB printer on /dev/usb/lp1<br>Configured as hand.</p></li><li><p>hewlett-packard hp-laserjet-4: USB printer on /dev/usb/lp0<br>Configured as mono.</p></li></ul>
